kube/src/api/typed.rs

```diff
@@ -339,3 +352,17 @@ impl<K> From<Api<K>> for Client {
         api.client
     }
 }
+
+#[test]
+fn api_default_ns() {
+    use k8s_openapi::core::v1::Pod;
+    let mut client = Client::try_default();
+    client.default_ns = "dev";
```

Comment:
> I have the Client, and it should have the `default_ns` set at this point.

`Client` doesn't have `default_ns` anymore.

https://github.com/clux/kube-rs/blob/c0c0c62edb27d6bdaffc4de51c7aeb378e073482/kube/src/client/mod.rs#L42-L45

And `crate::config::DEFAULT_NAMESPACE_MARKER` in `api` couples `config` and `api`.

I wrote hacky because I was just thinking of having `"DEFAULT_NS"` in two places with some comments about the other one. Then always replace this placeholder by changing `kube::Service::new` to decorate any service passed with `map_request`. It's not clean, but it should work.